### PR TITLE
chore: Add examples for API

### PR DESCRIPTION
**Pull request by vrongmeal**
These are the examples I am thinking of having for the API docs:

- [x] Create basin and get basin config
- [x] Reconfigure basin
- [x] Delete basin
- [x] List all basins: Fetch all basins taking `has_more` in consideration.
- [x] Create stream and get stream config
- [x] Reconfigure stream
- [x] Delete stream
- [x] List streams with prefix
- [x] Explicit trim: Delete data till latest sequence number
- [x] Producer (with Fencing token)
- [x] Get latest record
- [x] Consumer

### CODE REVIEW

**Review by infiniteregrets — COMMENTED** [review]
can you give `rustdoc-scrape-examples` a try here?

**Comment by infiniteregrets** [comment]
https://doc.rust-lang.org/rustdoc/scraped-examples.html

File: examples/get_latest_record.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let token = std::env::var("S2_AUTH_TOKEN")?;
+    let config = ClientConfig::new(token);
+    let basin: BasinName = "my-basin".parse()?;
+    let stream = "my-basin";
+    let stream_client = StreamClient::new(config, basin, stream);
+
+    let tail = stream_client.check_tail().await?;
+    let latest_seq_num = tail - 1;
```
Comment (shikhar) [inline examples/get_latest_record.rs:15]: tail will be 0 if no records on the stream

Comment (vrongmeal) [reply examples/get_latest_record.rs:15]: Oh yeah, I'll fix this.

**Review by shikhar — APPROVED** [review]
nice, thank you!

**Pull request merged**